src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -1165,3 +1165,80 @@ LFORTRAN_API void _lpython_close(int64_t fd)
         exit(1);
     }
 }
+
+#include <unwind.h>
+#include <link.h>
+
+static _Unwind_Reason_Code unwind_callback(struct _Unwind_Context *context,
+        void *vdata) {
+    struct Stacktrace *d = (struct Stacktrace *) vdata;
+    uintptr_t pc = _Unwind_GetIP(context);
+    if (pc != 0) {
+        pc--;
+        if (d->size < LCOMPILERS_MAX_STACKTRACE_LENGTH) {
+            d->pc[d->size] = pc;
+            d->size++;
+        } else {
+            // TODO: Do not know how to report an error here
+        }
+    }
+    return _URC_NO_REASON;
+}
+
+struct Stacktrace get_stacktrace_addresses()
+{
+    struct Stacktrace d;
+    d.size = 0;
+    _Unwind_Backtrace(unwind_callback, &d);
+    return d;
+}
+
+struct dl_phdr_info {
+    ElfW(Addr) dlpi_addr;
+    const char *dlpi_name;
+    const ElfW(Phdr) *dlpi_phdr;
+    ElfW(Half) dlpi_phnum;
+};
+
+extern int dl_iterate_phdr (int (*__callback) (struct dl_phdr_info *,
+    size_t, void *), void *__data);
```

Comment:
I added this because we get the following warning
```py
home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1208:32: warning: ‘struct dl_phdr_info’ declared inside parameter list will not be visible outside of this definition or declaration
 1208 | int shared_lib_callback(struct dl_phdr_info *info,
      |                                ^~~~~~~~~~~~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c: In function ‘shared_lib_callback’:
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1211:29: error: invalid use of undefined type ‘struct dl_phdr_info’
 1211 |     for (int i = 0; i < info->dlpi_phnum; i++) {
      |                             ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1212:17: error: invalid use of undefined type ‘struct dl_phdr_info’
 1212 |         if (info->dlpi_phdr[i].p_type == PT_LOAD) {
      |                 ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1213:39: error: invalid use of undefined type ‘struct dl_phdr_info’
 1213 |             ElfW(Addr) min_addr = info->dlpi_addr + info->dlpi_phdr[i].p_vaddr;
      |                                       ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1213:57: error: invalid use of undefined type ‘struct dl_phdr_info’
 1213 |             ElfW(Addr) min_addr = info->dlpi_addr + info->dlpi_phdr[i].p_vaddr;
      |                                                         ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1214:50: error: invalid use of undefined type ‘struct dl_phdr_info’
 1214 |             ElfW(Addr) max_addr = min_addr + info->dlpi_phdr[i].p_memsz;
      |                                                  ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1220:51: error: invalid use of undefined type ‘struct dl_phdr_info’
 1220 |                 d->local_pc = d->current_pc - info->dlpi_addr;
      |                                                   ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1221:37: error: invalid use of undefined type ‘struct dl_phdr_info’
 1221 |                 printf("%lx\n", info->dlpi_addr);
      |                                     ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1208:32: warning: ‘struct dl_phdr_info’ declared inside parameter list will not be visible outside of this definition or declaration
 1208 | int shared_lib_callback(struct dl_phdr_info *info,
      |                                ^~~~~~~~~~~~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c: In function ‘shared_lib_callback’:
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1211:29: error: invalid use of undefined type ‘struct dl_phdr_info’
 1211 |     for (int i = 0; i < info->dlpi_phnum; i++) {
      |                             ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1212:17: error: invalid use of undefined type ‘struct dl_phdr_info’
 1212 |         if (info->dlpi_phdr[i].p_type == PT_LOAD) {
      |                 ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1213:39: error: invalid use of undefined type ‘struct dl_phdr_info’
 1213 |             ElfW(Addr) min_addr = info->dlpi_addr + info->dlpi_phdr[i].p_vaddr;
      |                                       ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1213:57: error: invalid use of undefined type ‘struct dl_phdr_info’
 1213 |             ElfW(Addr) min_addr = info->dlpi_addr + info->dlpi_phdr[i].p_vaddr;
      |                                                         ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1214:50: error: invalid use of undefined type ‘struct dl_phdr_info’
 1214 |             ElfW(Addr) max_addr = min_addr + info->dlpi_phdr[i].p_memsz;
      |                                                  ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1220:51: error: invalid use of undefined type ‘struct dl_phdr_info’
 1220 |                 d->local_pc = d->current_pc - info->dlpi_addr;
      |                                                   ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1221:37: error: invalid use of undefined type ‘struct dl_phdr_info’
 1221 |                 printf("%lx\n", info->dlpi_addr);
      |                                     ^~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c: In function ‘print_stacktrace_addresses’:
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1235:13: warning: implicit declaration of function ‘dl_iterate_phdr’ [-Wimplicit-function-declaration]
 1235 |         if (dl_iterate_phdr(shared_lib_callback, &d) == 0) {
      |             ^~~~~~~~~~~~~~~
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c: In function ‘print_stacktrace_addresses’:
/home/thirumalai/Open_Source/lpython/src/libasr/runtime/lfortran_intrinsics.c:1235:13: warning: implicit declaration of function ‘dl_iterate_phdr’ [-Wimplicit-function-declaration]
 1235 |         if (dl_iterate_phdr(shared_lib_callback, &d) == 0) {
      |             ^~~~~~~~~~~~~~~
make[2]: *** [src/runtime/legacy/CMakeFiles/lpython_runtime_static.dir/build.make:76: src/runtime/legacy/CMakeFiles/lpython_runtime_static.dir/__/__/libasr/runtime/lfortran_intrinsics.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:477: src/runtime/legacy/CMakeFiles/lpython_runtime_static.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [src/runtime/legacy/CMakeFiles/lpython_runtime.dir/build.make:76: src/runtime/legacy/CMakeFiles/lpython_runtime.dir/__/__/libasr/runtime/lfortran_intrinsics.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:451: src/runtime/legacy/CMakeFiles/lpython_runtime.dir/all] Error 2
```